pkg/apis/maistra/v2/smcp_new.yaml

```diff
@@ -0,0 +1,410 @@
+      prometheus:
+        address: some-url-to-prometheus # specify this or install (i'm not sure if we can support this or not)
+        install: # install prometheus, as opposed to using an existing install at address
+          selfManaged: true # use prometheus CR, maybe poor naming?
+          config:
+            retention: 6h
+            scrapeInterval: 15s
+          useTLS: true # .Values.prometheus.security.enabled, i don't think this applies to us
+          service:
+            metadata: {}
+            nodePort: null # use node port if not null
+            ingress: # enables Ingress/Route if not null
+              contextPath: /prometheus
+              tls: # raw yaml, so it can be used with both Route and Ingress resources
+                termination: reencrypt
+          runtime:
+            deployment: {}
+            pod: {}
+
+    tracing:
+      type: Jaeger # or Zipkin, Lightstep, Datadog, Stackdriver
+      jaeger:
+        name: jaeger # name of Jaeger CR to reference/create
+        install: # create a jaeger CR if it doesn't already exist
+          config:
+            storage:
+              type: Memory # or Elasticsearch
+              memory: # implies tracing.jaeger.template=all-in-one
+                maxTraces: 100000
+              # or
+              elasticSearch: # implies tracing.jaeger.template=production-elasticsearch
+                nodeCount: 3
+                storage: {} # raw jaeger config, .Values.tracing.jaeger.elasticsearch.storage
+                redundancyPolicy: {} # raw jaeger config, .Values.tracing.jaeger.elasticsearch.redundancyPolicy
+                indexCleaner: {} # raw jaeger config, .Values.tracing.jaeger.elasticsearch.esIndexCleaner
+                runtime: {} # .Values.tracing.jaeger.elasticsearch.<runtime-specific-settings>
+          ingress: # configure Ingress/Route if not null
+            metadata:
+              labels: {}
+              annotations: {}
+          runtime: # similar to gateway.runtime.pod
+            containers:
+              default: {} # imageTag, imageRegistry, resources, etc.
+
+    visualization:
+      grafana:
+        address: some-grafana-url # use existing grafana install, or...
+        install: # install grafana if not null
+          selfManaged: true # use grafana CR, maybe poor naming?
+          config:
+            env: {} # .Values.grafana.env
+            envSecrets: {} # .Values.grafana.envSecrets
+          persistence:
+            storageClassName: ""
+            accessMode: ReadWriteOnce
+            capacity: 5Gi # not supported in charts
+          service: # similar to addons.tracing.jaeger.install.config.service
+            ingress:
+              contextPath: /grafana
+              tls:
+                termination: reencrypt
+          runtime:
+            deployment: {}
+            pod: {}
+      kiali:
+        name: kiali # kiali CR to create/reference
+        install: # install kiali CR if not present
```

Comment:
I don't think we should nest the options underneath `install`, because every addon setting we provide will be an installation setting. maybe `install` should just be the bool that determines whether the addon is managed?

Reply:
I'm trying to preserve the existing behavior, where the maistra operator installs and manages these things.  The difference is, once installed, maistra will only manage certain fields in the CR.  For example, in the case of Kiali, the settings for grafana, prometheus, and jaeger, and possibly the version (to ensure it is compatible with the control plane version).